core/src/main/java/org/jenkins/ui/symbol/Symbol.java

```diff
@@ -0,0 +1,111 @@
+    static final String PLACEHOLDER_MATCHER = "M368 368L144 144M368 144L144 368";
+
+    private Symbol() {}
+
+    /**
+     * Generates the svg markup for the given symbol name and attributes.
+     * @param request the symbol request object.
+     * @return The svg markup for the symbol.
+     * @since TODO
+     */
+    public static String get(@NonNull SymbolRequest request) {
+        String name = request.getName();
+        String title = request.getTitle();
+        String tooltip = request.getTooltip();
+        String classes = request.getClasses();
+        String pluginName = request.getPluginName();
+        String id = request.getId();
+        String translatedName = IconSet.cleanName(name);
+
+        String identifier = StringUtils.defaultIfBlank(pluginName, "core");
+
+        String symbol = SYMBOLS
+                .computeIfAbsent(identifier, key -> new ConcurrentHashMap<>())
+                .computeIfAbsent(translatedName, key -> loadSymbol(identifier, key));
+        if (StringUtils.isNotBlank(tooltip)) {
+            symbol = symbol.replaceAll("<svg", "<svg tooltip=\"" + Functions.htmlAttributeEscape(tooltip) + "\"");
+        }
+        if (StringUtils.isNotBlank(id)) {
+            symbol = symbol.replaceAll("<svg", "<svg id=\"" + Functions.htmlAttributeEscape(id) + "\"");
+        }
+        if (StringUtils.isNotBlank(classes)) {
+            symbol = symbol.replaceAll("<svg", "<svg class=\"" + Functions.htmlAttributeEscape(classes) + "\"");
+        }
+        if (StringUtils.isNotBlank(title)) {
+            symbol = "<span class=\"jenkins-visually-hidden\">" + Util.xmlEscape(title) + "</span>" + symbol;
+        }
+        return symbol;
+    }
+
+
+    @SuppressFBWarnings(value = {"NP_LOAD_OF_KNOWN_NULL_VALUE", "RCN_REDUNDANT_NULLCHECK_OF_NULL_VALUE"}, justification = "Spotbugs doesn't grok try-with-resources")
+    private static String loadSymbol(String namespace, String name) {
+        String markup = PLACEHOLDER_SVG;
+        ClassLoader classLoader = getClassLoader(namespace);
+        if (classLoader != null) {
+            try (InputStream inputStream = classLoader.getResourceAsStream("images/symbols/" + name + ".svg")) {
+                if (inputStream != null) {
+                    markup = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+                } else {
+                    LOGGER.log(Level.FINE, "Missing symbol " + name + " in " + namespace);
+                }
+            } catch (IOException e) {
+                LOGGER.log(Level.FINE, "Failed to load symbol " + name, e);
+            }
+        }
+        return markup.replaceAll("(<title>).*?(</title>)", "$1$2")
+                     .replaceAll("<svg", "<svg aria-hidden=\"true\"")
+                     .replaceAll("(class=\").*?(\")", "")
+                     .replaceAll("(tooltip=\").*?(\")", "")
+                     .replaceAll("(id=\").*?(\")", "")
+                     .replace("stroke:#000", "stroke:currentColor");
```

Comment:
@NotMyFault might this be the root cause of this PR instead? https://github.com/jenkinsci/jenkins/pull/6686

Reply:
`replace` replaces all, it's just that it's not a regex.

Reply:
> The dot of the price tag has no stroke defined, this replacement pattern wouldn't apply at all.

ah my bad.

it could be yes although I guess it's an assumption based on what ionicon's does

Reply:
Looks like everything is fine so far. However having some logic to validate symbols at build time would be interesting.

What comes to my mind:
* Check that each `path` has at least one of `fill`,`stroke` defined and set to either `none` or `currentColor`